### DISCUSSION TITLE
How to cancel uploading of emitted data

### DISCUSSION THREAD

**Discussion by iokarkan** [miguelgrinberg/python-socketio#1352] [General]
I have an app that is constantly streaming data via websockets. I have noticed that during low-bandwidth network conditions the emit queue is growing as uploads take too long, and the system RAM ends up getting full.

I thought timeout errors in call would also mean that the corresponding upload is stopped, but this seems not to be the case.

Is there any way to stop uploads in-flight via socketio, and if not what is the recommended way of handling this situation?

**Comment by miguelgrinberg** [comment]
The JavaScript Socket.IO has the "volatile" option that may help in this situation, but this isn't currently available for Python. At some point I considered adding and never got around to do it.

With the current version you can have the client acknowledge events as it receives them, so that the server can pace its sending to the rate at which the client is receiving.
